ERCS/erc-7965.md

```diff
@@ -0,0 +1,184 @@
+type: Standards Track
+category: ERC
+created: 2025-06-06
+requires: 7786
+---
+
+## Abstract
+
+This document defines standardized broadcasting semantics and attributes for [ERC-7786] cross-chain messaging that enable trustless message verification through cryptographic proofs. Messages are committed on source chains in verifiable ways, then verified on destination chains using cryptographic proofs of the source chain's state or transaction history. This approach provides cross-chain communication without relying on external validators or bridge operators.
+
+[ERC-7786]: ./eip-7786.md
+
+## Motivation
+
+Cross-chain messaging protocols typically rely on external validators, multisigs, or optimistic mechanisms that introduce trust assumptions and potential points of failure. Cryptographic proofs offer an alternative approach where messages can be verified using the consensus mechanisms and cryptographic commitments of the chains themselves (e.g. storage proofs for EVM chains).
+
+However, cryptographic proof verification requires chain-specific routing information, proof data, and verification parameters that are not addressed by the base [ERC-7786] interface. Additionally, [ERC-7786] does not explicitly define broadcasting semantics for cross-chain messaging. Broadcasting enables messages to be sent without specifying a particular receiver, making them available for verification and execution by any contract on destination chains.
+
+This document standardizes these requirements as [ERC-7786] attributes, enabling cryptographic proof-based messaging within the established cross-chain messaging framework. The specification supports multi-hop verification paths, allowing messages to traverse through multiple intermediary chains when direct verification is not possible.
+
+The key benefits of this approach include:
+
+- **Trustless verification**: No external validators or multisigs required. Chains trust their own consensus mechanisms.
+- **Universal compatibility**: Works between chains with verifiable state relationships through shared settlement infrastructure or compatible proof systems.
+- **Flexible messaging patterns**: Supports both targeted and broadcast messaging through standardized semantics, enabling new classes of applications like oracles and intent settlement systems.
+- **Composability**: Full integration with existing [ERC-7786] infrastructure and tooling
+
+## Specification
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119 and RFC 8174.
+
+### Broadcasting Semantics
+
+Gateways MUST interpret an empty string `""` in the receiver address of `sendMessage()` as broadcasting. This convention allows cryptographic proof gateways to support both targeted messaging (to specific receivers) and broadcast messaging (to any interested party) within the same interface.
```

Comment:
With ERC-7786 using ERC-7930 interoperable address, this translates to something like

"Gateway MUST accept ERC-7930 interoperable addresses that have an empty address component as valid recipient when sending a message."